test/test_rgb2ycbcr.py

```diff
@@ -0,0 +1,182 @@
+                rgb.data_valid.next = True
+
+                for i in range(samples):
+
+                    # rgb signal assignment in the dut
+                    rgb.red.next = inputs['red'][i]
+                    rgb.green.next = inputs['green'][i]
+                    rgb.blue.next = inputs['blue'][i]
+
+                    if ycbcr.data_valid == 1:
+                        for ycbcr_act, val in zip(('y', 'cb', 'cr'),
+                                                  (int(ycbcr.y), int(ycbcr.cb),
+                                                   int(ycbcr.cr))):
+
+                            actual_outputs[ycbcr_act].append(val)
+
+                    yield clock.posedge
+
+                print_results(inputs, expected_outputs, actual_outputs)
+                raise StopSimulation
+
+            return tbdut, tbclk, tbstim
+
+        inst = bench_color_trans()
+        inst.config_sim(trace=True)
+        inst.run_sim()
+
+    else:
+
+        exp_y = tuple(expected_outputs['y'])
+        exp_cb = tuple(expected_outputs['cb'])
+        exp_cr = tuple(expected_outputs['cr'])
+
+        in_r = tuple(inputs['red'])
+        in_g = tuple(inputs['green'])
+        in_b = tuple(inputs['blue'])
+
+        y_s, cb_s, cr_s = [Signal(intbv(0)[pixel_bits:])
+                           for _ in range(3)]
+
+        @myhdl.block
+        def bench_color_trans():
+            tbdut = rgb2ycbcr(rgb, ycbcr, clock, reset, num_fractional_bits)
+            tbclk = clock_driver(clock)
+            tbrst = rstonstart(reset, clock)
+
+            @instance
+            def tbstim():
+                yield reset.negedge
+                rgb.data_valid.next = True
+
+                for i in range(samples):
+                    # rgb signal assignment in the dut
+                    rgb.red.next = in_r[i]
+                    rgb.green.next = in_g[i]
+                    rgb.blue.next = in_b[i]
+
+                    if ycbcr.data_valid == 1:
+
+                        # expected_outputs signal assignment
+                        y_s.next = exp_y[i-3]
+                        cb_s.next = exp_cb[i-3]
+                        cr_s.next = exp_cr[i-3]
+                        yield delay(1)
+                        print("Expected outputs ===>Y:%d Cb:%d Cr:%d"
+                              % (y_s, cb_s, cr_s))
+                        print("Actual outputs ===>Y:%d Cb:%d Cr:%d"
+                              % (ycbcr.y, ycbcr.cb, ycbcr.cr))
+                        print("----------------------------")
+                    yield clock.posedge
+
+                raise StopSimulation
+
+            return tbdut, tbclk, tbstim, tbrst
+
+        # verify and convert with iverilog
+        verify.simulator = 'iverilog'
+        assert bench_color_trans().verify_convert() == 0
+        # verify and convert with GHDL
+        verify.simulator = 'ghdl'
+        assert bench_color_trans().verify_convert() == 0
+
+
+def test_color_translation():
```

Comment:
I have been thinking about this a little bit.  I think I understand your motivation to combine the two tests into one (code reuse?) but we loose report summary.  When the functional test and the conversion test were separate `py.test` would provide information which failed.  Now that they have been combined, if the test fails we will have to spend time parsing the logs to determine which test failed, vs. having the information immediately.

```
$ py.test test_rgb2ycbcr.py
============================= test session starts =============================
Python 3.5.1, pytest-2.9.2, py-1.4.31, pluggy-0.3.1
collected 2 items

test_rgb2ycbcr.py ..

========================== 2 passed in 0.99 seconds ===========================
```

I know this seems minor at this point but as our test suite grows having a quick summary of 
which tests failed is useful.  The more granular each test (i.e. individual tests that test specific features) the easier it will be to debug as the project grows.

@mkatsimpris could you comment why you merged the functional and conversion test?